include/roaring/roaring.h

```diff
@@ -904,85 +904,114 @@ typedef struct roaring_uint32_iterator_s {
- * values. If there is a  value, then this iterator points to the first value
- * and `it->has_value` is true. The value is in `it->current_value`.
+ * Initialize an iterator object that can be used to iterate through the values.
+ * If there is a  value, then this iterator points to the first value and
+ * `it->has_value` is true. The value is in `it->current_value`.
  */
-void roaring_init_iterator(const roaring_bitmap_t *r,
+void roaring_iterator_init(const roaring_bitmap_t *r,
                            roaring_uint32_iterator_t *newit);
 
+/** DEPRECATED, use `roaring_iterator_init`. */
+#define roaring_init_iterator roaring_iterator_init
+
 /**
- * Initialize an iterator object that can be used to iterate through the
- * values. If there is a value, then this iterator points to the last value
- * and `it->has_value` is true. The value is in `it->current_value`.
+ * Initialize an iterator object that can be used to iterate through the values.
+ * If there is a value, then this iterator points to the last value and
+ * `it->has_value` is true. The value is in `it->current_value`.
  */
-void roaring_init_iterator_last(const roaring_bitmap_t *r,
+void roaring_iterator_init_last(const roaring_bitmap_t *r,
                                 roaring_uint32_iterator_t *newit);
 
+/** DEPRECATED, use `roaring_iterator_init_last`. */
+#define roaring_init_iterator_last roaring_iterator_init_last
+
 /**
  * Create an iterator object that can be used to iterate through the values.
  * Caller is responsible for calling `roaring_free_iterator()`.
  *
- * The iterator is initialized (this function calls `roaring_init_iterator()`)
+ * The iterator is initialized (this function calls `roaring_iterator_init()`)
  * If there is a value, then this iterator points to the first value and
  * `it->has_value` is true.  The value is in `it->current_value`.
  */
-roaring_uint32_iterator_t *roaring_create_iterator(const roaring_bitmap_t *r);
+roaring_uint32_iterator_t *roaring_iterator_create(const roaring_bitmap_t *r);
+
+/** DEPRECATED, use `roaring_iterator_create`. */
+#define roaring_create_iterator roaring_iterator_create
 
 /**
  * Advance the iterator. If there is a new value, then `it->has_value` is true.
  * The new value is in `it->current_value`. Values are traversed in increasing
  * orders. For convenience, returns `it->has_value`.
  *
- * Once `it->has_value` is false, `roaring_advance_uint32_iterator` should not
- * be called on the iterator again. Calling `roaring_previous_uint32_iterator`
+ * Once `it->has_value` is false, `roaring_uint32_iterator_advance` should not
+ * be called on the iterator again. Calling `roaring_uint32_iterator_previous`
  * is allowed.
  */
-bool roaring_advance_uint32_iterator(roaring_uint32_iterator_t *it);
+bool roaring_uint32_iterator_advance(roaring_uint32_iterator_t *it);
+
+/** DEPRECATED, use `roaring_uint32_iterator_advance`. */
+#define roaring_advance_uint32_iterator roaring_uint32_iterator_advance
```

Comment:
We could use a `static inline` function in these places, so we can try to use `__attribute__ ((deprecated))` if we can? Not sure if it's worth it.

And/or, it might be nice to have a `ROARING_NO_DEPRECATED` which users can define to make sure they don't accidentally use anything deprecated.

Reply:
We rely on C++11 in CRoaring right? I believe it was added later.

Reply:
gcc has an extension (`__attribute__((deprecated))`) that clang also supports (since at least gcc 4.5 and clang 4.0, the farthest back I bothered to look). MSVC appears to also have a `__declspec(deprecated)`, but I'm not sure how to check what version of MSVC added it, I'd be fine with the gcc check only (which should also cover clang I think) and default to expanding to nothing.

Reply:
Done (for GCC / clang).